File: String/TypeScript/_0709_ToLowerCase.ts
/**
 * Iteration
 * Time Complexity: BigO(N)
 * Space Complexity: BigO(N)
 */
function toLowerCase(s: string): string {
  const str = s.split("");
  for (let i = 0; i < str.length; i++) {
    if (str[i].charCodeAt(0) > 64 && str[i].charCodeAt(0) < 91)
      str[i] = String.fromCharCode(str[i].charCodeAt(0) + 32);
  }
  return str.join("");
}

console.log(toLowerCase("Hello"));
console.log(toLowerCase("here"));
console.log(toLowerCase("LOVELY"));

File: String/Python/_0709_ToLowerCase.py
class Solution:
    # Iteration
    # Time Complexity: BigO(N)
    # Space Complexity: BigO(N)
    def toLowerCase(self, s: str) -> str:
        res = ""
        for c in s:
            # ord(): find unicode of 'c' (inverse of ord(): chr())
            if ord("A") <= ord(c) <= ord("Z"):
                res += chr((ord(c) - ord("A")) + ord("a"))
            else:
                res += c
        return res


ins = Solution()

print(ins.toLowerCase("Hello"))
print(ins.toLowerCase("here"))
print(ins.toLowerCase("LOVELY"))
